### DISCUSSION TITLE
Material Web vanilla JS port

### DISCUSSION THREAD

**Discussion by dRhaoma** [material-components/material-web#5166] [General]
Hi

Since I work more or less with vanilla JS, and the Material Lib uses React, typescript and scss, I was wondering about the possibility for a vanilla JS port, or more specifically wondering about the rules surrounding a custom partial port of the project? Hereby meaning partially rewriting the typescript and React elements to suit vanilla javascript. Is that allowed?

Regards

**Comment by asyncliz** [comment]
Hi! This Material library does not use React, and does not require Typescript or Sass to use. You can use it in plain old javascript with or without frameworks.

[jsbin example](https://jsbin.com/siqecativo/edit?html,output)

```html
<!DOCTYPE html>
<html>
<head>
  <title>Material Web JS</title>
  <style>
    body {
      display: flex;
      gap: 16px;
      align-items: center;
      flex-wrap: wrap;
    }
  </style>
  <script type="module">
    import 'https://esm.run/@material/web/all.js';
  </script>
</head>
<body>
  <md-filled-button>Filled</md-filled-button>
  <md-outlined-button>Outlined</md-outlined-button>
  <md-checkbox></md-checkbox>
  <md-switch></md-switch>
  <md-radio name="radio"></md-radio>
  <md-radio name="radio"></md-radio>
  <md-radio name="radio"></md-radio>
</body>
</html>
```

**Reply by dRhaoma** [reply]
Thanks, never felt quite so dumb. I can't understand how if haven't stumbled across this

**Comment by JaviToro** [comment]
Hi!

First of all, thank you for this thread and the reply, but I wasn't unable to find it in the documentation either. Maybe is something so simple, but I think it'd be nice to add it to the docs.

Thanks.
